### ISSUE TITLE
Issues when zone sensor is the controller

### ISSUE THREAD

**Issue by epicurei** [zxdavb/evohome#53]
- 1) as you can check in the attached file the zone "00" is reported in "Schema[01:095571 (evohome)]" with sensor "null" instead of with "01:095571" because it has been assigned as the current temperature that of the sensor on the controller and the same to 
|| CTL:095571 | HGI:201230 | RP | zone_devices     | 0004 || {'zone_type': '04', 'zone_idx': '00', 'device_role': 'zone_sensor', 'devices': []}

- 2) in "Status[01:095571 (evohome)]" correctly in the zone "00" is reported the temperature value of 23.19 that the controller "01:095571" sends to the zone "00" together with the setpoint as detectable in the first two lines of the attachment, but which is not found in any other parameter as the sensor on the controller 01:095571
[client summary evohome controller.txt](https://github.com/zxdavb/ramses_rf/files/9463511/client.summary.evohome.controller.txt)

**Comment by zxdavb** [comment]
> I will take this opportunity to ask all who read this to provide packet logs, not the formatted output from **client.py**.

@epicurei In the post above, you do not explicitly state your issue, nor do you describe the behaviour that you expect... but I believe I can still answer your question.

TL;DR: The behaviour you describe is as expected. If required, workarounds exist.

----------------------------

The **ramses_rf** library uses *discovery* (i.e. `000C` commands) to determine the system schema (e.g. which device is the sensor for a zone).

If the controller is configured as the sensor for a zone, then it will say that the zone has no sensor (nor does it broadcast its temperature readings).

Similarly: if the `appliance_control` device (a.k.a boiler relay) is an OpenTherm bridge (rather than a simple relay, BDR91), then it (the controller) will say there is not such device configured.

Obviously, I cannot change the behaviour of the controller, but you can use *eavesdropping* to determine the above data:
```
python client.py -e monitor /dev/ttyACM0
```

In principle, and certqinly for the Heat (CH/DHW) domain, *eavesdropping* shoudl be avoided. Instead, use a hand-crafted schema:
```
python client.py -c config.json monitor /dev/ttyACM0
```

... an example of such a file:
```
{
    "config": {
        "enable_discovery": true,
        "enable_eavesdrop": false,
    },
    "01:123456": {
        "system": {
            "appliance_control": "10:888888"
        },
        "zones": {
            "01": {"sensor": "01:123456"}
        }
    }
}
```

**Comment by epicurei** [comment]
[evohome controllers configurations v0.21.06.txt](https://github.com/zxdavb/ramses_rf/files/9912597/evohome.controllers.configurations.v0.21.06.txt)
[message trace v0.21.6 -.txt](https://github.com/zxdavb/ramses_rf/files/9912598/message.trace.v0.21.6.-.txt)
[packet.log](https://github.com/zxdavb/ramses_rf/files/9912599/packet.log)

11:48:21.862  I --- 13:103760 --:------ 13:103760 3EF0 003 0000FF < Support the development of ramses_rf by reporting this packet (byte 5: )

11:48:39.094  I --- 13:107116 --:------ 13:107116 3EF0 003 00C8FF < Support the development of ramses_rf by reporting this packet (byte 5: )

15:17:41.881  I --- 04:065911 --:------ 01:095571 3150 002 03CA < Coding error: ValueError(Invalid result: 1.01 (0xCA) is > 1)
Traceback (most recent call last):
  File "/home/supervisor/Scrivania/ramses_rf-master v0.21.6 01_11 mod/ramses_rf/protocol/message.py", line 331, in _validate
    result = PAYLOAD_PARSERS.get(self.code, parser_unknown)(
  File "/home/supervisor/Scrivania/ramses_rf-master v0.21.6 01_11 mod/ramses_rf/protocol/parsers.py", line 141, in wrapper
    result = fnc(payload, msg, **kwargs)
  File "/home/supervisor/Scrivania/ramses_rf-master v0.21.6 01_11 mod/ramses_rf/protocol/parsers.py", line 2057, in parser_3150
    return valve_demand(payload[2:])  # TODO: check UFC/FC is == CTL/FC
  File "/home/supervisor/Scrivania/ramses_rf-master v0.21.6 01_11 mod/ramses_rf/protocol/helpers.py", line 27, in wrapper
    return fnc(*args, **kwargs)
  File "/home/supervisor/Scrivania/ramses_rf-master v0.21.6 01_11 mod/ramses_rf/protocol/helpers.py", line 328, in valve_demand
    raise ValueError(f"Invalid result: {result} (0x{value}) is > 1")
ValueError: Invalid result: 1.01 (0xCA) is > 1
